vllm/worker/model_runner.py

```diff
@@ -851,29 +897,52 @@ def _maybe_cupy_nccl():
         yield
 
 
+def _pad_to_alignment(x: List[int], multiple_of: int, pad: int) -> List[int]:
+    return x + [pad] * ((-len(x)) % multiple_of)
+
+
 def _pad_to_max(x: List[int], max_len: int, pad: int) -> List[int]:
     assert len(x) <= max_len
     return x + [pad] * (max_len - len(x))
 
 
+def _align_if_necessary(x: List[int], pad: int, should_align=SHOULD_ALIGN):
+    """Align flattened 1D inputs by a fixed alignment size."""
+    if not should_align:
+        return x
+    batch_size = len(x)
+    batch_size = _get_graph_batch_size(batch_size)
+    return _pad_to_alignment(x, batch_size, pad)
+
+
 def _make_tensor_with_pad(
     x: List[List[int]],
     max_len: int,
     pad: int,
     dtype: torch.dtype,
     device: Optional[Union[str, torch.device]],
 ) -> torch.Tensor:
+    """Make a padded tensor of a 2D inputs.
+
+    The padding is applied to the end of each inner list until it reaches
+    `max_len`.
+    """
     padded_x = [_pad_to_max(x_i, max_len, pad) for x_i in x]
     return torch.tensor(padded_x, dtype=dtype, device=device)
 
 
 def _get_graph_batch_size(batch_size: int) -> int:
+    """Returns the padded batch size given actual batch size.
+
+    Batch sizes are 1, 2, 4, _BATCH_SIZE_ALIGNMENT,
+    2*_BATCH_SIZE_ALIGNMENT, 3*_BATCH_SIZE_ALIGNMENT...
+    """
     if batch_size <= 2:
         return batch_size
     elif batch_size <= 4:
         return 4
     else:
-        return (batch_size + 7) // 8 * 8
+        return _get_aligned_size(batch_size, _BATCH_SIZE_ALIGNMENT)
```

Comment:
Again, I don't think we need a separate helper function here because the `(x + align - 1) // align * align` pattern is pretty common and simple.